File: architect-agent/app/contracts/chat_interface.py
from enum import Enum
from datetime import datetime
from typing import Optional, Union
from pydantic import BaseModel


class UserIntent(str, Enum):
    plan = "plan"
    accept = "accept"
    refine = "refine"
    undefined = "undefined"


class NodeName(str, Enum):
    intent = "intent_node"
    solution = "solution_node"
    solution_review = "solution_review_node"
    plan = "plan_node"
    plan_review = "plan_review_node"
    reply = "reply_node"


class ChatStatus(str, Enum):
    is_active = "isActive"
    is_stopped = "isStopped"


class AgentStatus(str, Enum):
    is_thinking = "isThinking"
    has_replied = "hasReplied"


class ChatActor(str, Enum):
    user = "User"
    agent = "Agent"


class FeatureInterface(BaseModel):
    feature: str


class ComponentInterface(BaseModel):
    tech: str
    features: list[FeatureInterface] = []


class SolutionInterface(BaseModel):
    architecture: str
    components: list[ComponentInterface] = []


class RequirementInterface(BaseModel):
    requirement: str


class AcceptanceCriterionInterface(BaseModel):
    criterion: str


class EpicInterface(BaseModel):
    id: str
    name: str
    requirements: list[RequirementInterface] = []
    solution: SolutionInterface


class TicketInterface(BaseModel):
    id: str
    epicId: str
    name: str
    requirements: list[RequirementInterface] = []
    acceptance_criteria: list[AcceptanceCriterionInterface] = []


class ReplyInterface(BaseModel):
    epic: EpicInterface
    tickets: list[TicketInterface] = []


class FinalReplyInterface(BaseModel):
    epicId: str
    ticketIds: list[str] = []


class MessageInterface(BaseModel):
    actor: ChatActor
    content: Union[str, ReplyInterface, FinalReplyInterface]
    agentStatus: Optional[AgentStatus] = None
    timestamp: Optional[datetime] = None
    node: Optional[NodeName] = None


class ChatInterface(BaseModel):
    id: str
    title: Optional[str] = None
    messages: list[MessageInterface] = []
    status: ChatStatus
    agentStatus: Optional[AgentStatus] = None


class ChatRequest(BaseModel):
    conversationId: str
    message: str
    history: list[MessageInterface] = []

File: backend/src/chat/contracts/chat.interface.ts
export enum ChatStatus {
  isActive = 'isActive',
  isStopped = 'isStopped',
}

export enum AgentStatus {
  isThinking = 'isThinking',
  hasReplied = 'hasReplied',
}

export enum ChatActor {
  user = 'User',
  agent = 'Agent',
}

export interface FeatureInterface {
  feature: string;
}

export interface ComponentInterface {
  tech: string;
  features: FeatureInterface[];
}

export interface SolutionInterface {
  architecture: string;
  components: ComponentInterface[];
}

export interface RequirementInterface {
  requirement: string;
}

export interface AcceptanceCriterionInterface {
  criterion: string;
}

export interface EpicInterface {
  id: string;
  name: string;
  requirements: RequirementInterface[];
  solution: SolutionInterface;
}

export interface TicketInterface {
  id: string;
  epicId: string;
  name: string;
  requirements: RequirementInterface[];
  acceptance_criteria: AcceptanceCriterionInterface[];
}

export interface ReplyInterface {
  epic: EpicInterface;
  tickets: TicketInterface[];
}

export interface FinalReplyInterface {
  epicId: string;
  ticketIds: string[];
}

export interface MessageInterface {
  actor: ChatActor;
  timestamp: Date;
  content: string | ReplyInterface | FinalReplyInterface;
  agentStatus?: AgentStatus | null;
}

export interface ChatInterface {
  id: string;
  title: string;
  timestamp?: Date;
  messages: MessageInterface[];
  status: ChatStatus;
  agentStatus?: AgentStatus;
  created_at?: Date;
  modified_at?: Date;
}

export interface ChatHistoryInterface {
  id: string;
  title?: string;
  createdAt: Date;
}

export interface ChatCreatedResponse {
  id: string;
}

export interface ChatContinueResponse {
  accepted: true;
}

export interface ChatStopResponse {
  stopped: true;
}